### PR TITLE
🧪 Improve Pagination tests and boundary logic

### PR DESCRIPTION
**Pull request by splk3**
This PR improves the test suite for the `Pagination` component and enhances its robustness.

### 🎯 What:
Addressed missing unit tests and improved existing component logic.

### 📊 Coverage:
- **Interaction simulation:** Now uses `@testing-library/user-event` instead of `fireEvent`.
- **Boundary logic:** Updated `Pagination.tsx` to use inequality checks for the `disabled` prop on buttons.
- **Edge cases:** Added tests for when `currentPage` is out of bounds (e.g., `currentPage > totalPages` or `currentPage < 1`).
- **Callback safety:** Verified that `onPageChange` is never called when buttons are disabled.

### ✨ Result:
Improved test reliability and component robustness against unexpected state values.

---
*PR created automatically by Jules for task [12705718550643353860](https://jules.google.com/task/12705718550643353860) started by @splk3*

### CODE REVIEW

**Comment by splk3** [comment]
@copilot resolve the merge conflicts in this pull request

File: src/components/Pagination.tsx [hunk 1]
```diff
@@ -40,7 +40,7 @@ export default function Pagination({ currentPage, totalPages, onPageChange }: Pa
 
       <button
         onClick={handleNext}
-        disabled={currentPage === totalPages}
+        disabled={currentPage >= totalPages}
```
[the remark this answers is not in the document]
Comment (splk3) [reply src/components/Pagination.tsx]: @copilot apply changes based on [this feedback](https://github.com/splk3/goalie-gen/pull/383#discussion_r3238630890)

File: src/components/__tests__/Pagination.test.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  });
+
+  it("disables the next button and does not call onPageChange if currentPage exceeds totalPages", async () => {
+    const user = userEvent.setup();
+    render(<Pagination currentPage={6} totalPages={5} onPageChange={onPageChangeMock} />);
+    const nextButton = screen.getByRole("button", { name: /next page/i });
+    expect(nextButton).toBeDisabled();
+    await user.click(nextButton);
+    expect(onPageChangeMock).not.toHaveBeenCalled();
+  });
```
[the remark this answers is not in the document]
Comment (splk3) [reply src/components/__tests__/Pagination.test.tsx]: @copilot apply changes based on the comments in this thread

**Comment by splk3** [comment]
@copilot apply changes based on the comments in [this thread](https://github.com/splk3/goalie-gen/pull/383#pullrequestreview-4293732299)
